core/src/main/scala/org/http4s/parser/SimpleHeaders.scala

```diff
@@ -95,12 +96,15 @@ private[parser] trait SimpleHeaders {
 //  // Do not accept scoped IPv6 addresses as they should not appear in the Host header,
 //  // see also https://issues.apache.org/bugzilla/show_bug.cgi?id=35122 (WONTFIX in Apache 2 issue) and
 //  // https://bugzilla.mozilla.org/show_bug.cgi?id=464162 (FIXED in mozilla)
-  def HOST(value: String): ParseResult[Host] = new Http4sHeaderParser[Host](value) {
-    def entry = rule {
-      (Token | IPv6Reference) ~ OptWS ~
-        optional(":" ~ capture(oneOrMore(Digit)) ~> (_.toInt)) ~ EOL ~> (Host(_:String, _:Option[Int]))
-    }
-  }.parse
+  def HOST(value: String): ParseResult[Host] =
+    new Http4sHeaderParser[Host](value) with Rfc3986Parser {
+      def charset = StandardCharsets.UTF_8
```

Comment:
This seems like a bad hardcoding, but, geez, is anybody sending these that aren't UTF-8?